### ISSUE TITLE
Trilinos Framework:  Create Dev->Master build on Vortex

### ISSUE THREAD

**Issue by prwolfe** [trilinos/Trilinos#8129]
## Enhancement
@trilinos/framework 

request #7047 is being moved to run on vortex.

Current status:

A set of environment and config files has been created (the branches at https://github.com/prwolfe/Trilinos/tree/vortex_build)

Builds with c++14 at -j 44 in about 4 hours with all packages.

tests disabled in the cuda 9.2 build are still disabled
there were 38 test failures in 50 minutes

```
Total Test time (real) = 2958.00 sec

The following tests FAILED:
	608 - Zoltan_ch_simple_zoltan_parallel (Failed)
	611 - Zoltan_hg_simple_zoltan_parallel (Failed)
	1382 - SEACASAprepro_lib_aprepro_lib_unit_test (Failed)
	1383 - SEACASAprepro_lib_aprepro_lib_array_test (Failed)
	1385 - SEACASAprepro_aprepro_unit_test (Failed)
	1386 - SEACASAprepro_aprepro_array_test (Failed)
	1387 - SEACASAprepro_aprepro_command_line_vars_test (Failed)
	1388 - SEACASAprepro_aprepro_command_line_include_test (Failed)
	1389 - SEACASAprepro_aprepro_test_dump_reread (Failed)
	2920 - ROL_example_PDE-OPT_ginzburg-landau_example_02_MPI_4 (Failed)
	2935 - Piro_AnalysisDriverTpetra_MPI_4 (Failed)
	3080 - PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-2 (Failed)
	3081 - PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-3 (Failed)
	3082 - PanzerAdaptersSTK_CurlLaplacianMultiblockExample-ConvTest-Quad-Order-1 (Failed)
	3084 - PanzerAdaptersSTK_MixedCurlLaplacianExample-ConvTest-Quad-Order-1 (Failed)
	3085 - PanzerAdaptersSTK_MixedCurlLaplacianExample-ConvTest-Quad-Order-2 (Failed)
	3086 - PanzerAdaptersSTK_MixedCurlLaplacianExample-ConvTest-Quad-Order-3 (Failed)
	3087 - PanzerAdaptersSTK_MixedCurlLaplacianMultiblockExample-ConvTest-Quad-Order-1 (Failed)
	3088 - PanzerAdaptersSTK_MixedCurlLaplacianExample-ConvTest-Tri-Order-2 (Failed)
	3089 - PanzerAdaptersSTK_MixedPoissonExample (Failed)
	3090 - PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-1 (Failed)
	3091 - PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-2 (Failed)
	3096 - PanzerAdaptersSTK_PoissonInterfaceTpetra_2d_MPI_4 (Failed)
	3098 - PanzerAdaptersSTK_PoissonInterfaceTpetra_3d_MPI_4 (Failed)
	3099 - PanzerAdaptersSTK_PoissonInterfaceTpetra_2d_MPI_4-ConvTest (Failed)
	3100 - PanzerAdaptersSTK_PoissonInterfaceTpetra_3d_MPI_4-ConvTest (Failed)
	3101 - PanzerAdaptersSTK_assembly_example_MPI_4 (Failed)
	3102 - PanzerAdaptersSTK_main_driver_energy-ss (Failed)
	3103 - PanzerAdaptersSTK_main_driver_energy-transient (Failed)
	3104 - PanzerAdaptersSTK_main_driver_energy-ss-blocked (Failed)
	3105 - PanzerAdaptersSTK_main_driver_energy-ss-loca-eigenvalue (Failed)
	3108 - PanzerAdaptersSTK_main_driver_energy-transient-blocked (Failed)
	3113 - Adelus_vector_random_MPI_1 (Failed)
	3114 - Adelus_vector_random_MPI_2 (Failed)
	3115 - Adelus_vector_random_MPI_3 (Failed)
	3116 - Adelus_vector_random_MPI_4 (Failed)
	3138 - TrilinosCouplings_fenl_ensemble_8x8x8_belos_muelu_MPI_4 (Failed)
	3139 - TrilinosCouplings_fenl_pce_8x8x8_belos_muelu_mean_based_MPI_4 (Timeout)
Errors while running CTest
```

And a test of CUDA_LAUNCH+BLOCKING=0 is currently underway.

**Comment by prwolfe** [comment]
@jwillenbring - a new issue to track this work.

**Comment by prwolfe** [comment]
Looked closer at the timing data - c++14 build was 3h 28m

**Comment by prwolfe** [comment]
the CUDA_LAUNCH_BLOCKING=0 test completed. Compilation time was identical to the c++14 test (this also used c++14) but almost 100% of tests failed.


>
>4% tests passed, 3030 tests failed out of 3146
>

**Comment by rppawlo** [comment]
@MicheldeMessieres - any chance you could take a look at the errors here with launch blocking disabled? Hoping it is just some simple configure issue. @kddevin

**Comment by kddevin** [comment]
@rppawlo :  Michel's contract is ended; we'll have to investigate on our own.
@prwolfe : was your second bullet above (regarding Panzer and TrilinosCoupling) with CUDA_LAUNCH_BLOCKING=0?  Do they exhibit the same behavior with CUDA_LAUNCH_BLOCKING=1?
@prwolfe :  is there a way that we can see the output from your runs (i.e., is there a link on the dashboard)?  Looking at that output might be faster than my trying to reproduce the Zoltan errors.  Thanks!

**Comment by prwolfe** [comment]
@kddevin The Panzer and TrilinosCouplings tests show the same behavior with CUDA_LAUNCH_BLOCKING=1 and CUDA_LAUNCH_BLOCKING=0. I have been running by hand so there is no dashboard line (that is my next step and likely to happen tomorrow.) I will post a link here when that is available.

**Comment by kddevin** [comment]
@prwolfe Thanks for confirming the results with/without CUDA_LAUNCH_BLOCKING.  
For the Zoltan failures, I'd be happy to just look at Testing/Temporary/LastTest.log in your build directory if that's easier.  You can email the path to me if you want to take that approach.  Thanks.

**Comment by prwolfe** [comment]
I went to set up the Jenkins job for this and ran into 2 issues

1) the queue had been closed on Friday. Looks like a reboot over the weekend resolved this.
2) lsf is copyist the shell script we use for our top-level startup. That fails as it expects to find other scripts in the same directory it starts from. I am looking at those fixes now.

Paul

**Comment by william76** [comment]
I worked on some of the issues Paul was seeing with the Vortex build in a PR to his fork. Details are here:
- https://github.com/prwolfe/Trilinos/pull/15

**Comment by jwillenbring** [comment]
@prwolfe summarized the state of this yesterday saying that he has a build working in his account, but it does not yet run properly through the Trilinos account.

**Comment by prwolfe** [comment]
Status: Builds run as prwolfe, but when run as the trilinos entity through Jenkins fail during configure. I am working to replicate that build on the command line and check which resources are failing, but the machine has been down or busy the past day.

**Comment by prwolfe** [comment]
Still seeing queue submissions hanging and have submitted tickets to vortex-help.

**Comment by bartlettroscoe** [comment]
@trilinos/framework, @e10harvey 

FYI: If you are serious about setting up a Trilinos PR build on 'vortex', note that we are still seeing random mass failures as described in #6861 nd recently discussed in:

* https://servicedesk.sandia.gov/servicedesk/customer/portal/4/ONESTOP-16877

These seem to be occurring less frequently than they used to but ATS-2 'vortex' is a moving target so you can expect problems.

**Comment by prwolfe** [comment]
Found an environment issue which when resolved allowed me to run and submit to cash - see https://testing.sandia.gov/cdash/build/8403204. That is empty as the PR changes no code, but I will set that to build and report full builds until we get them cleaned up. Then we will move it to PR's

**Comment by prwolfe** [comment]
Regular builds of the entire set of Trilinos Packages are not set to run at 3 am and 3pm - the turn around is 5 hours so we should see reports at 8am and 8pm

Tests failed in the initial run are

ROL_example_poisson-inversion_example_01_MPI_1
Adelus_vector_random_MPI_1
SEACASAprepro_lib_aprepro_lib_array_test
SEACASAprepro_aprepro_command_line_vars_test
SEACASAprepro_aprepro_array_test
SEACASAprepro_aprepro_test_dump_reread
SEACASAprepro_aprepro_unit_test
SEACASAprepro_lib_aprepro_lib_unit_test
SEACASAprepro_aprepro_command_line_include_test
Adelus_vector_random_MPI_2
Adelus_vector_random_MPI_3
Adelus_vector_random_MPI_4
Zoltan_ch_simple_zoltan_parallel
Zoltan_hg_simple_zoltan_parallel
KokkosCore_UnitTest_CudaTimingBased_MPI_1
Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests_MPI_1

For which we will need to file issues. These need to be either fixed, deemed unneeded, or otherwise set to return Passed before we can use this build.

**Comment by prwolfe** [comment]
Changing title to reflect the fact that we are going to push this build to dev->master promotion rather than as a PR build

**Comment by prwolfe** [comment]
Status as of 3/5/2021

#8819 was merged to resolve #8801 - test has not run since.

#8829 was merged trying to fix the SECAS tests - that had a problem that is fixed in #8841. That is not mergeable at this point as ride is down.

No activity from the Adelus, Zoltan, ROL, or Kokkos teams on their issue.

**Comment by kddevin** [comment]
For Zoltan, see #8798 and #8995 ; zoltan issues are as resolved as they can be until SpectrumMPI is fixed on vortex.  Thanks for your patience, @prwolfe.

**Comment by prwolfe** [comment]
> For Zoltan, see #8798 and #8995 ; zoltan issues are as resolved as they can be until SpectrumMPI is fixed on vortex. Thanks for your patience, @prwolfe.

Thanks! I will turn those off in the build when I get a chance. Might be a bit as the ECP Annual Meeting is this week.

**Comment by jwillenbring** [comment]
With #9449 merged, what is left here? Just turning on a dev->master build?

**Comment by jwillenbring** [comment]
This has been running successfully. Closing.
